src/GitHub.VisualStudio/TeamFoundation/TeamFoundationResolver.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using System.Diagnostics.CodeAnalysis;
+using System.ComponentModel.Composition;
+
+namespace GitHub.VisualStudio.TeamFoundation
+{
+    [Export]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public sealed class TeamFoundationResolver : IDisposable
```

Comment:
This and the other new MEF exports should be exporting interfaces and not the actual types, for easier testing.